test/unit/local/steps/initial_diff.js

```diff
@@ -0,0 +1,100 @@
+/* eslint-env mocha */
+/* @flow */
+
+const should = require('should')
+const Builders = require('../../../support/builders')
+const configHelpers = require('../../../support/helpers/config')
+const pouchHelpers = require('../../../support/helpers/pouch')
+
+const Buffer = require('../../../../core/local/steps/buffer')
+const initialDiff = require('../../../../core/local/steps/initial_diff')
+
+describe('local/steps/initialDiff', () => {
+  let builders
+  let buffer
+
+  before('instanciate config', configHelpers.createConfig)
+  beforeEach('instanciate pouch', pouchHelpers.createDatabase)
+  beforeEach('populate pouch with documents', function () {
+    builders = new Builders({pouch: this.pouch})
+    buffer = new Buffer()
+  })
+  afterEach('clean pouch', pouchHelpers.cleanDatabase)
+  after('clean config directory', configHelpers.cleanConfig)
+
+  it('detects documents moved while client was stopped', async function () {
+    builders.metadir().path('foo').ino(1).create()
+    builders.metafile().path('baz').ino(2).create()
+
+    const bar = builders.event().action('scan').kind('directory').path('bar').stats({ ino: 1 }).build()
+    const buz = builders.event().action('scan').kind('file').path('buz').stats({ ino: 2 }).build()
+    buffer.push([bar, buz])
+    buffer = initialDiff(buffer, { pouch: this.pouch })
+
+    const events = await buffer.pop()
+    should(events).deepEqual([
+      builders.event(bar).action('renamed').oldPath('foo').build(),
+      builders.event(buz).action('renamed').oldPath('baz').build()
+    ])
+  })
+
+  it('detects documents moved while client is doing initial scan', async function () {
+    builders.metadir().path('foo').ino(1).create()
+    builders.metafile().path('foo/baz').ino(2).create()
+    builders.metadir().path('bar').ino(3).create()
+
+    const foo = builders.event().action('scan').kind('directory').path('foo').stats({ ino: 1 }).build()
+    const barbaz = builders.event().action('created').kind('file').path('bar/baz').stats({ ino: 2 }).build()
+    buffer.push([foo, barbaz])
+    const bar = builders.event().action('scan').kind('directory').path('bar').stats({ ino: 3 }).build()
+    buffer.push([
+      bar,
+      builders.event().action('scan').kind('file').path('bar/baz').stats({ ino: 2 }).build()
```

Comment:
nitpicking: some events are inline, others are not.

Reply:
I inline them when I can but start putting them on multiple lines when they go beyond 80 characters (I've been a bit more lenient about this recently and only do it if they go way beyond that limit).

Do you have a different approach?

Reply:
I either inline everything or nothing, I find it easier to scan since everything looks the same.

Reply:
Possibly just a matter of taste & not a blocker anyway.